misk-cron/src/main/kotlin/misk/cron/CronCoordinator.kt

```diff
@@ -0,0 +1,30 @@
+package misk.cron
+
+import jakarta.inject.Inject
+import wisp.lease.LeaseManager
+
+interface CronCoordinator {
+  fun shouldRunTask(taskName: String): Boolean
+}
+
+class SingleLeaseCronCoordinator @Inject constructor(
+  private val leaseManager: LeaseManager
+) : CronCoordinator {
+  override fun shouldRunTask(taskName: String): Boolean {
+    val lease = leaseManager.requestLease(CRON_CLUSTER_LEASE_NAME)
+    return lease.checkHeld() || lease.acquire()
+  }
+
+  companion object {
+    const val CRON_CLUSTER_LEASE_NAME = "misk.cron.lease"
+  }
+}
+
+class DistributedCronCoordinator @Inject constructor(
```

Comment:
Maybe a better name is `MultipleLeaseCronCoordinator` to avoid confusion with distributed leases